File: src/main/java/com/github/thorben/webnd/application/tech/Message.java
package com.github.thorben.webnd.application.tech;

public class Message {

	private String key;
	private String content;
	private int uses;

	public Message(String key, String content, int uses) {
		this.key = key;
		this.content = content;
		this.uses = uses;
		if(uses == 0) {
			throw new IllegalArgumentException("The amount of uses has to be -1 or greater then 0");
		}
	}

	public Message(String key, String content) {
		this(key, content, -1);
	}

	public static Message once(String key, String value) {
		return new Message(key, value, 1);
	}

	public String getContent() {
		if(uses > 0) {
			uses -= 1;
		}

		return content;
	}

	public boolean valid() {
		return uses != 0;
	}

	public String getKey() {
		return key;
	}
}

File: src/main/kotlin/com/github/thorben/webnd/application/tech/Message.kt
package com.github.thorben.webnd.application.tech

class Message(
		val key: String,
		private val content: String,
		private var uses: Int = -1
) {

	fun getContent(): String {
		if (uses > 0) {
			uses -= 1
		}
		return content
	}

	fun valid(): Boolean {
		return uses != 0
	}

	companion object {
		fun once(key: String, value: String): Message {
			return Message(key, value, 1)
		}
	}

	init {
		require(uses != 0) { "The amount of uses has to be -1 or greater then 0" }
	}
}